From f362ae624c6fc46a8a2f183dda2e6a68f4e41f35 Mon Sep 17 00:00:00 2001
From: Philip Chimento <philip.chimento@gmail.com>
Date: Tue, 25 Nov 2014 11:28:26 -0500
Subject: [PATCH] build: Use m4 directory for macros

Previously, ACLOCAL_AMFLAGS was set to ${ACLOCAL_FLAGS}. This caused
problems when running libtoolize --force --copy as part of autoreconf.
On JHBuild when ACLOCAL_FLAGS was set to the build tree's share/aclocal
directory, libtoolize would actually remove the libtool macros from
there.

https://bugzilla.gnome.org/show_bug.cgi?id=740696
---
 .gitignore   | 1 +
 Makefile.am  | 2 +-
 configure.ac | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index 85de3e8..3f203a7 100644
--- a/.gitignore
+++ b/.gitignore
@@ -31,6 +31,7 @@ gjs_gi_probes.h
 install-sh
 libtool
 ltmain.sh
+/m4
 missing
 stamp-h1
 test_user_data
diff --git a/Makefile.am b/Makefile.am
index 4dbd318..ff2c008 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -14,7 +14,7 @@ EXTRA_DIST =
 TEST_PROGS =
 check_PROGRAMS = $(TEST_PROGS)
 INTROSPECTION_GIRS =
-ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS}
+ACLOCAL_AMFLAGS = -I m4
 AM_CFLAGS = $(WARN_CFLAGS) -DG_LOG_DOMAIN=\"Gjs\"
 AM_CPPFLAGS = $(WARN_CXXFLAGS) -DG_LOG_DOMAIN=\"Gjs\"
 
diff --git a/configure.ac b/configure.ac
index 6efd58d..d6d0a86 100644
--- a/configure.ac
+++ b/configure.ac
@@ -11,6 +11,7 @@ AC_PREREQ(2.61)
 AC_INIT([gjs], pkg_version, [http://bugzilla.gnome.org/enter_bug.cgi?product=gjs])
 AM_INIT_AUTOMAKE([dist-xz no-dist-gzip])
 AC_CONFIG_SRCDIR([gjs/console.cpp])
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADER([config.h])
 
 AC_DEFINE([GJS_VERSION], pkg_int_version, [The gjs version as an integer])
-- 
1.9.3 (Apple Git-50)

